examples/object_detection/ios/ObjectDetector/ViewContollers/DetectorViewController.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import MediaPipeTasksVision
+
+/**
+ * Base class to view controllers that handle performing detection on images, videos or camera frames.
+ */
+class DetectorViewController: UIViewController {
```

Comment:
I think we should have less number of classes so that the code would become easier to read for developers who came to this sample.

Therefore, I don't think we should have this base class but fold the feature implemented here into `CameraViewController` and `MediaLibraryViewController` because:
* DetectorMetadata observer is only used in CameraViewController
* `draw` method doesn't do anything meaningful. It just pass a list of parameters to overlayView for drawing.